From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jbicha@debian.org>
Date: Wed, 12 Nov 2025 22:29:09 -0500
Subject: build: Enable all builtin loaders

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index f478326..09e069a 100644
--- a/meson.build
+++ b/meson.build
@@ -265,6 +265,7 @@ endif
 
 # Check which loaders should be built into gdk-pixbuf
 builtin_loaders = get_option('builtin_loaders')
+enabled_loaders += builtin_loaders
 if builtin_loaders.contains('default')
   if glycin_dep.found()
     builtin_loaders = ['glycin']
