From 140a0b7d511589647256795a05c3b114f226a6af Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Wed, 18 Dec 2019 08:53:28 +0100
Subject: [PATCH] RDA: Disable switch user button, if session is remote.

---
 src/gs-lock-plug.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/src/gs-lock-plug.c
+++ b/src/gs-lock-plug.c
@@ -1236,6 +1236,12 @@
 
 	if (switch_enabled)
 	{
+#ifdef HAVE_RDA
+		if (rda_session_is_remote()) {
+			gs_debug ("No switch user capability if session is remote");
+			gtk_widget_hide (plug->priv->auth_switch_button);
+		} else
+#endif /* HAVE_RDA */
 		if (process_is_running ("mdm"))
 		{
 			/* MDM  */
